app/facades/acapy_wallet.py

```diff
@@ -60,7 +60,13 @@ async def create_did(controller: AcaPyClient) -> Did:
     return Did(did=did_result.result.did, verkey=did_result.result.verkey)
 
 
-async def set_public_did(controller: AcaPyClient, did: str) -> DID:
+async def set_public_did(
+    controller: AcaPyClient,
+    did: str,
+    connection_id: str = None,
+    create_transaction_for_endorser: bool = False,
+    is_endorsed: bool = False,
```

Comment:
This feels a bit misleading. What happens if is_endorsed is True and there is no result.result? there won't be a did returned right (which the return type of this method says it does). Maybe we can add an union return type that also takes into account the endorsement return type. This way we don't have to pass is_endorsed which sometimes can hard to know

Reply:
If it is endorsed the set public did succeeds but no result.result is returned (in 0.7.4)

Reply:
So this prevents the throwing an error when result.result is not present

Reply:
Yes, but the return type is misleading in that case. Could we maybe trigger it based on the return status code instead in that case? (So if no error is thrown we see it as being succeeded). is_endorsed requires deep knowledge of the setup, which may not always be known.

Reply:
ah controller doesn't return the status code if i'm not mistaken. I'll think of something else